### PR TITLE
Allow StakingAdmin to set `min_commission`

### PR DESCRIPTION
**Pull request by Ank4n**
## Changes
- Rename `SlashCancelOrigin` to `AdminOrigin`.
- New call `set_min_commission` which can be called by `AdminOrigin`. 

polkadot companion: https://github.com/paritytech/polkadot/pull/6444.
Related to https://github.com/paritytech/substrate/issues/12930.
Old PR: https://github.com/paritytech/substrate/pull/12943.

### CODE REVIEW

**Comment by Ank4n** [comment]
/cmd queue -c bench-bot $ pallet dev pallet_staking

File: frame/staking/src/pallet/mod.rs [hunk 1]
```diff
@@ -1792,6 +1793,20 @@ pub mod pallet {
 			})?;
 			Ok(())
 		}
+
+		/// Sets the minimum amount of commission that each validators must maintain.
+		///
+		/// This call has lower privilege requirements than `set_staking_config` and can be called
+		/// by the `T::AdminOrigin`. Root can always call this.
+		///
+		/// Weight: `O(1)`
```
Comment (rossbulat) [inline frame/staking/src/pallet/mod.rs:1802]: remove this weight comment?

File: frame/staking/src/pallet/mod.rs [hunk 2]
```diff
@@ -1792,6 +1793,20 @@ pub mod pallet {
...
+		#[pallet::call_index(25)]
+		#[pallet::weight(T::WeightInfo::set_min_commission())]
+		pub fn set_min_commission(origin: OriginFor<T>, new: Perbill) -> DispatchResult {
```
Comment (rossbulat) [inline frame/staking/src/pallet/mod.rs:1805]: Name this `apply_min_commission` to keep it inline with `force_apply_min_commission`?

File: frame/staking/src/tests.rs [hunk 3]
```diff
@@ -5726,6 +5726,46 @@ fn scale_validator_count_errors() {
 	})
 }
 
+#[test]
+fn set_min_commission_works_with_admin_origin() {
+	ExtBuilder::default().build_and_execute(|| {
+		// no minimum commission set initially
+		assert_eq!(MinCommission::<Test>::get(), Perbill::from_percent(0));
```
Comment (rossbulat) [inline frame/staking/src/tests.rs:5733]: nitpick, use Perbill::zero() ?

File: frame/staking/src/tests.rs [hunk 4]
```diff
@@ -5726,6 +5726,46 @@ fn scale_validator_count_errors() {
...
+
+		// root can set min commission
+		assert_ok!(Staking::set_min_commission(RuntimeOrigin::root(), Perbill::from_percent(10),));
```
Comment (rossbulat) [inline frame/staking/src/tests.rs:5736]: remove the comma at `(10),));`

**Review by rossbulat — COMMENTED** [review]
Just some syntax clean up and one query about `set_min_commission` function. I'll also test a bit later.

File: frame/staking/src/pallet/mod.rs (hunk 1 above)
Comment (kianenigma) [reply frame/staking/src/pallet/mod.rs:1802]: yes please.

File: frame/staking/src/tests.rs (hunk 4 above)
Comment (kianenigma) [inline frame/staking/src/tests.rs:5736]:
```suggestion
		assert_ok!(Staking::set_min_commission(RuntimeOrigin::root(), Perbill::from_percent(10)));
```

File: frame/staking/src/tests.rs (hunk 3 above)
Comment (kianenigma) [reply frame/staking/src/tests.rs:5733]: or `Zero::zero()`, which will work if you move from `Perbill` to `Permill` or others :))

**Review by kianenigma — APPROVED** [review]
Ho ho ho! 🎅 LGTM, except good comments from @rossbulat

File: frame/staking/src/pallet/mod.rs (hunk 2 above)
Comment (Ank4n) [reply frame/staking/src/pallet/mod.rs:1805]: The `force_apply_min_commission` applies `min_commission` on the provided validator which is different thing than setting the value right? Other calls more similar to this are `set_validator_count` and `set_staking_config`.

**Review by rossbulat — APPROVED** [review]

**Comment by Ank4n** [comment]
bot merge

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-37/1736/1

**Pull request merged**